src/plugins/workspace/public/plugin.ts

```diff
@@ -73,7 +78,7 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
     /**
      * Retrieve workspace id from url
      */
-    const workspaceId = this.getWorkspaceIdFromURL();
+    const workspaceId = this.getWorkspaceIdFromURL() || PUBLIC_WORKSPACE;
```

Comment:
I'm wondering if we need to make `PUBLIC_WORKSPACE` a default workspace id here? For public workspace, it should have its own url like `/w/public` which can be retrieved from url as well.

Reply:
Agree, change the logic here.